secrets/src/int/secret_integers.rs

```diff
@@ -0,0 +1,235 @@
+use super::classify::*;
+use crate::traits::*;
+use std::ops::*;
+
+pub type I8 = Secret<i8>;
+pub type U8 = Secret<u8>;
+pub type I16 = Secret<i16>;
+pub type U16 = Secret<u16>;
+pub type I32 = Secret<i32>;
+pub type U32 = Secret<u32>;
+pub type I64 = Secret<i64>;
+pub type U64 = Secret<u64>;
+pub type I128 = Secret<i128>;
+pub type U128 = Secret<u128>;
+
+// Construct a secret integer
+pub const fn secret<T>(x: T) -> Secret<T> {
+    Secret(x)
+}
+
+impl<T: Scalar> Classify for T {
+    type Classified = Secret<T>;
+    fn classify(self) -> Secret<Self> {
+        secret(self)
+    }
+}
+
+impl<T: Scalar> Declassify for Secret<T> {
+    type Declassified = T;
+    fn declassify(self) -> T {
+        self.0
+    }
+}
+
+impl<'a, T: Scalar> ClassifyRef for &'a T {
+    type ClassifiedRef = &'a Secret<T>;
+    fn classify_ref(self) -> &'a Secret<T> {
+        unsafe { core::mem::transmute(self) }
+    }
+}
+
+impl<'a, T: Scalar> DeclassifyRef for &'a Secret<T> {
+    type DeclassifiedRef = &'a T;
+    fn declassify_ref(self) -> &'a T {
+        unsafe { core::mem::transmute(self) }
+    }
+}
+
+pub fn classify_mut_slice<T: Scalar>(x: &mut [T]) -> &mut [Secret<T>] {
```

Comment:
Should this be a trait?

Reply:
Added a comment: we would like hax to be able to process this function and it is easier if it were a function than a trait method.